pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -390,6 +395,24 @@ func (r *ReconcileUserSignup) generateCompliantUsername(instance *toolchainv1alp
 	return "", NewSignupError(fmt.Sprintf("unable to transform username [%s] even after 100 attempts", instance.Spec.Username))
 }
 
+func transformUsername(username string) string {
+	userName := specialCharRegexp.ReplaceAllString(strings.Split(username, "@")[0], "-")
```

Comment:
It's actually possible to create a user with username starting with `@`. I've just checked it. In this case the username will be an empty string if I'm not missing anything. Let's cover this case too.
For example we could transform `@spiderman` to `crt-spiderman` or to `at-spiderman`. I like `at-spiderman` better but `crt-spiderman` would work too.

Reply:
WWIW, I prefer the `at-` prefix (or `-at` suffix)

Reply:
Thanks for pointing this out - I missed this case.
done: 5b4422c